apps/web/src/views/BuyCrypto/hooks/usePriceQuoter.ts

```diff
@@ -68,49 +67,61 @@ const usePriceQuotes = (amount: string, inputCurrency: string, outputCurrency: s
   const [quotes, setQuotes] = useState<ProviderQoute[]>([])
   const { chainId } = useActiveChainId()
 
-  const fetchProviderAvailability = async (ip: string, combinedData: ProviderQoute[]) => {
-    // first check user availability
-    const responsePromises = [fetchMoonpayAvailability(ip), fetchMercuryoAvailability(ip)]
-    const responses = await Promise.allSettled(responsePromises)
+  const fetchProviderAvailability = useCallback(
+    async (ip: string, combinedData: ProviderQoute[]) => {
+      // first check user availability
+      const responsePromises = [fetchMoonpayAvailability(ip), fetchMercuryoAvailability(ip)]
+      const responses = await Promise.allSettled(responsePromises)
 
-    const dataPromises = responses.reduce((accumulator, response) => {
-      if (response.status === 'fulfilled') {
-        return [...accumulator, response.value]
-      }
-      console.error('Error fetching price quotes:', response.reason)
-      return accumulator
-    }, [])
+      const dataPromises = responses.reduce((accumulator, response) => {
+        if (response.status === 'fulfilled') {
+          return [...accumulator, response.value]
+        }
+        console.error('Error fetching price quotes:', response.reason)
+        return accumulator
+      }, [])
 
-    const [moonPayAvailability, mercuryoAvailability] = await Promise.all(dataPromises)
+      const [moonPayAvailability, mercuryoAvailability] = await Promise.all(dataPromises)
```

Comment:
I think we don't need await here? The values in `dataPromises` are already resolved

Reply:
Another question, what if the first fetch failed and the second fetch succeeded? Isn't `moonPayAvailability` becoming `mercuryoAvailability`? I think in this case we should use `map` instead of `reduce` above